### PR TITLE
Add a "shock touch" mutation

### PR DESCRIPTION
**Pull request by BalthazarArgall**
#### Summary
Content "Add a 'shock touch' mutation"

#### Purpose of change
I'm looking to add a new mutation line (or maybe "subline") to the game, while I'm still in the process of making different ideas coalesce into a fully coherent line, at this point I'm pretty sure I want to build it around the ability to stun stuff through an eel-like shocking touch which doesn't yet exists in the game.

#### Describe the solution
I created a "Shock Touch" mutation along with the spell, EOCs and precursor mutation support it.

Now, science.

The prerequisite to any electric ability is some way to produce current, this is what electrocytes are for, they're modified muscle cells that can produce ~0.15v when they polarize. 

I am assuming enough electrocytes to produce a power potential of 800v which is somewhat high but not the maximum observed voltage produced by electric eels.

Shocks delivered by eels are made up of 2ms pulses delivered at 100hz. We have to assume a resistance applicable to most enemies in the game, I think 500ohms is fair given the shock will be delivered through touch with a mucus coating and the fact that during strong enough shocks skin breaks, lowering the resistance further, a lot of enemies also have already broken skin.

While in reality eels don't deliver shocks during a set amount of time but rather keep shocking until tired or their objective is accomplished, I had to choose a "standard shock duration" of 1 second.

The total energy spent per shock is hence (800v squared/500ohms) * 0.002 seconds * 100 pulses = 256J

256J = ~60 gram calories so that's the amount of calories needed to recover 1 shock "charge".

I couldn't find information on how many shocks can be delivered "back-to-back", I found unsupported claims that eels can deliver 6 shocks in a minute or that they can spend almost a minute shocking a crocodile to kill it so I'll make the maximum charge worth 10 shocks and 10 minutes to recover 1 charge, I welcome more evidence based figures if you have them.

TLDR, how does it work in game?

You first mutate electrocytes, which converts 60 calories to 1 charge each 10 minutes up to a maximum of 10 charges. This, however doesn't do anything on its own.

Then you mutate shock touch, which represent both the external visible mutation made to deliver the shock and the internal ability to polarize your electrocytes.

This gives you the "Shock Touch" active ability, when you activate it you enter aim mode, choose a target and either cancel or go through with the shock. Shocking a target takes 1 second and applies the "zapped" status effect for between 3 and 5 seconds, provided the target can be zapped. This consumes 1 electrocytes charge.

#### Describe alternatives you've considered
A loooot of stuff to mention here.
1. Make stuff conditional with the size of the target, either the duration of the status, the amount of charges consumed or the possibility of a shock in the first place. This is, to my admittedly limited knowledge, impossible or at least very hard to do given how the mutation/spells are set up. 
2. Somehow limbify the mutation/spell, I kept the actual physical mechanics of the shock application intentionally vague, this is because there's at least 3 different ways to make a circuit with a target just for humanoids, and what if you don't even have 4 limbs or they're tentacles?
3. Have some kind of combo where the shock first downs the target then if you shock a downed target it gets stunned. Same issue as point 1, I just don't know how I could achieve that.
4. Implement a more elegant way to spend electrocytes charge, at the moment I use a modified "ink glands" structure where I increment/decrement an effect to keep track and expose to the player the amount of charges he gets but this comes with the issue that **cancelling while in aiming mode still consumes a charge.** I have absolutely no idea how to fix this given the current architecture but I welcome any idea for a solution.

#### Testing
Plenty, unless I broke something since testing everything should work, except for a "bug" that I mentioned in point 4 above.

#### Additional context
Implementation and research has been done with the help of @Kamejeir.

### CODE REVIEW

**Review by Kamejeir — CHANGES_REQUESTED** [review]
alright, fleshing it out

**Review by Kamejeir — COMMENTED** [review]
mending those discussed things

File: data/json/mutations/mutation_spells.json [hunk 1]
```diff
@@ -184,5 +184,39 @@
...
     "damage_type": "biological"
+  },
+  {
+    "id": "shock_touch",
+    "type": "SPELL",
+    "name": { "str": "Shock touch" },
+    "description": "Shocks an enemy.",
+    "effect": "attack",
+    "TODO": "Decide how to properly implement the shock.",
+    "effect_str": "blind",
```
[the remark this answers is not in the document]
Comment (BalthazarArgall) [reply data/json/mutations/mutation_spells.json:195]: Why not stunned too?

**Review by Kamejeir — COMMENTED** [review]
Still need to find out how to make it combo into itself. At least it seems reapplying effects is not necessary (somewhat nulling my point about not including stun), so maybe it's as simple as calling for one shock_touch variant that zaps if not downed, and one that zaps + downs otherwise.

**Comment by BalthazarArgall** [comment]
Oops, forgot to reinstall vcpkg

**Comment by Kamejeir** [comment]
assuming you refer to the windows build failing at vcpkg, it has irked me too throughout seemingly random commits. I would guess it is not going to matter - the merger kicked that test until that part was successful

**Comment by BalthazarArgall** [comment]
> assuming you refer to the windows build failing at vcpkg, it has irked me too throughout seemingly random commits. I would guess it is not going to matter - the merger kicked that test until that part was successful

Uh, you're right, it worked this time, that's the first time this test flakes on me.

**Comment by kevingranade** [comment]
This MUST
Not work as well on larger targets. 
Not work as well or more likely at all on targets without exposed skin. This includes clothes and tougher exteriors like chitin or even hide.
Require you to physically attack/grapple the target, a "blast" is wildly innapropriate.

**Comment by BalthazarArgall** [comment]
> This MUST Not work as well on larger targets. Not work as well or more likely at all on targets without exposed skin. This includes clothes and tougher exteriors like chitin or even hide. Require you to physically attack/grapple the target, a "blast" is wildly innapropriate.

I'll make it work the same way as bite attacks then, this should give it a more appropriate behavior. It means the shocks will be dealt randomly instead of having a deliberate usage though.

**Comment by Night-Pryanik** [comment]
Closing as stale. If you wish to continue working on this, ping me to reopen.
